packages/headless/src/composables/slider/createSlider.ts

```diff
@@ -0,0 +1,861 @@
+import { computed, onBeforeUnmount, ref, unref, watch, type MaybeRef, type Ref } from "vue";
+import { areArraysEqual } from "../../utils/array.js";
+import { createBuilder, createElRef } from "../../utils/builder.js";
+import { isFocusVisible } from "../../utils/dom.js";
+import { MathUtils } from "../../utils/math.js";
+import type { Nullable } from "../../utils/types.js";
+
+export type SliderMark =
+  | {
+      value: number;
+      label?: string;
+    }
+  | number;
+
+export type SliderOrientation = "horizontal" | "vertical";
+
+export type CreateSliderOptions = {
+  /**
+   * Current value(s) of the slider.
+   * Each value should be between min and max.
+   *
+   * @default [min]
+   */
+  values: Ref<number[]>;
+  /**
+   * Minimum value of the slider.
+   *
+   * @default 0
+   */
+  min: MaybeRef<number>;
+  /**
+   * Maximum value of the slider.
+   *
+   * @default 100
+   */
+  max: MaybeRef<number>;
+  /**
+   * Step size for the slider.
+   *
+   * @default 1
+   */
+  step: MaybeRef<number | null>;
+  /**
+   * Step size when holding shift key.
+   *
+   * @default 10
+   */
+  shiftStep?: MaybeRef<number>;
+  /**
+   * Whether the slider is disabled.
+   *
+   * @default false
+   */
+  disabled?: MaybeRef<boolean>;
+  /**
+   * Array of marks to display on the slider.
+   *
+   * @default false
+   */
+  marks?: MaybeRef<Nullable<SliderMark[] | boolean>>;
+  /**
+   * Aria label for the slider.
+   *
+   * @default undefined
+   */
+  label?: MaybeRef<string>;
```

Comment:
```suggestion
  /**
   * Aria label for the slider.
   */
  label: MaybeRef<string>;
```

The label should be required for accessibility reasons :)